### ISSUE TITLE
Stack count or compare to null?

### ISSUE THREAD

**Issue by aidinzolghadr** [prime31/RecyclerKit#3]
Hi,

In TrashManRecycleBin's line number 214, you compare return value of Peek with null but I face an error with this line saying that "Operation is not valid due to the current state of the object" and it's thrown when the _gameObjectPool is empty.

I think we should instead compare it's count property against zero. I trust your C# skills are far better than mine so I'm not sure abut this and probably I'm missing something but doing this edit apparently solved me problem.

So to sum it up, I think:
while( _gameObjectPool.Peek() != null )
should change to:
while( _gameObjectPool.Count > 0 )

Thanks.

**Comment by prime31** [comment]
That should indeed be a Count and not a Peek. It will still work but with
an exception on the last operation. The next push will have the change.


Mike
